app/controllers/texts_controller.rb

```diff
@@ -11,9 +11,13 @@ def show
   private
 
   def set_texts
-    genre = params[:genre]
+    if params[:q]
```

Comment:
```suggestion
    if params[:q]["genre"].present?
```
こっちで書くべきですね！

今のままでは条件式と真の場合の処理の内容の整合性が取り切れていないです！